Automattic/wp-calypso
# Import test/e2e into the monorepo

#### Changes proposed in this Pull Request

* Drop `test/e2e` package-lock.json, it will be part of root's `yarn.lock` now
* Import `test/e2e` as a subpackage in the monorepo

#### Testing instructions

1) Checkout the branch. Go to `test/e2e` and run `yarn` to install deps -> it should install all deps even without doing `yarn` in the root first.
2) Run the tests after installing deps. Go to `test/e2e` and run `./run.sh -g -x`. Verify that the tests start and work.

#### Performance

It is worth mentioning this will make installation times slower.


|  | Master  | This PR   | This PR + dedupe |
|---|---|---|---|
| Install deps in root | 1:03.88 | 1:47.50 | 1:35.17 |
| Installing deps in test/e2e with root deps already installed | 9.359 | 1.984 | 1.890 |
| Installing deps in test/e2e without root deps | 9.477 | 1:12.96 | 1:02.45 |

Note that this change will introduce some duplication in `yarn.lock`. Future PRs will remove that duplication, making the installation times better (3rd column in the table above).

#### e2e tests

There are two repos that know how to build calypso e2e tests: `wp-e2e-tests-for-branches` and `wp-e2e-tests-canary` and create CI jobs associated with calypso PRs. Those CI jobs *will fail* once this lands in master.

To fix them, we need to change how those two repos build e2e tests. The PRs with the change are already up:
* https://github.com/Automattic/wp-e2e-tests-for-branches/pull/57
* https://github.com/Automattic/wp-e2e-tests-canary/pull/25

Once this change lands in master, I'll update the git submodule in those two branches to point back to calypso master, and merge them. That should restore e2e.

#### Bundle sizes
(edit)

Bundle sizes will change a bit here and there because some module paths are different now (eg: `../node_modules/lib` to `./node_modules/lib`). It should only be a few bytes.
